ethportal-peertest/src/scenarios/find.rs

```diff
@@ -1,109 +1,98 @@
+    let store_result = peertest
+        .bootnode
+        .ipc_client
+        .store(content_key.clone(), content_value.clone())
+        .await
+        .unwrap();
+    assert!(store_result);
 
     // Send trace recursive find content request
-    let request = JsonRpcRequest {
-        method: "portal_historyTraceRecursiveFindContent".to_string(),
-        id: 12,
-        params: Params::Array(vec![json!(uniq_content_key)]),
+    let result = peertest.nodes[0]
+        .ipc_client
+        .trace_recursive_find_content(content_key)
+        .await
+        .unwrap();
+    let received_content_value = match result.content {
+        PossibleHistoryContentValue::ContentPresent(c) => c,
+        PossibleHistoryContentValue::ContentAbsent => panic!("Expected content to be found"),
     };
-
-    let result = make_ipc_request(&peertest.nodes[0].web3_ipc_path, &request).unwrap();
-    assert_eq!(result["content"], json!(HISTORY_CONTENT_VALUE.to_string()));
+    assert_eq!(received_content_value, content_value);
     assert_eq!(
-        result["route"],
-        json!([{"enr": &peertest.bootnode.enr.to_base64(), "distance": "0x100"}])
+        result.route,
+        vec![NodeInfo {
+            enr: peertest.bootnode.enr.clone(),
+            distance: U256::from_str_radix("0x100", 16).unwrap()
+        }]
     );
 }
 
 // This test ensures that when content is not found the correct response is returned.
-pub fn test_trace_recursive_find_content_for_absent_content(
-    _peertest_config: PeertestConfig,
-    peertest: &Peertest,
-) {
+pub async fn test_trace_recursive_find_content_for_absent_content(peertest: &Peertest) {
     // Different key to other test (final character).
     let uniq_content_key = "0x0015b11b918355b1ef9c5db810302ebad0bf2544255b530cdce90674d5887bb287";
+    let content_key: HistoryContentKey = serde_json::from_value(json!(uniq_content_key)).unwrap();
     // Do not store content to offer in the testnode db
 
     // Send trace recursive find content request
-    let request = JsonRpcRequest {
-        method: "portal_historyTraceRecursiveFindContent".to_string(),
-        id: 12,
-        params: Params::Array(vec![json!(uniq_content_key)]),
-    };
+    let result = peertest
+        .bootnode
+        .ipc_client
+        .trace_recursive_find_content(content_key)
+        .await
+        .unwrap();
 
-    let result = make_ipc_request(&peertest.nodes[0].web3_ipc_path, &request).unwrap();
-    assert_eq!(result["content"], json!(CONTENT_ABSENT.to_string()));
+    if let PossibleHistoryContentValue::ContentPresent(_) = result.content {
+        panic!("Expected content to be absent");
+    }
     // Check that at least one route was involved.
-    assert!(!result["route"].as_array().unwrap().is_empty());
-}
-// This test ensures that the jsonrpc channels don't close if there is an invalid recursive find
-// content request
-pub fn test_recursive_find_content_invalid_params(
```

Comment:
This scenario is no longer necessary since the jsonrpsee framework manages invalid content params